app/src/main/java/com/douglasbruce/tasky/core/domain/validation/PasswordValidator.kt

```diff
@@ -0,0 +1,27 @@
+package com.douglasbruce.tasky.core.domain.validation
+
+import javax.inject.Inject
+
+class PasswordValidator @Inject constructor() {
+
+    private val passwordRegex = Regex("^(?=.*\\d)(?=.*[a-z])(?=.*[A-Z]).*\$")
```

Comment:
I'm also a friend of avoiding regex whenever it's easy and makes sense. It's just unreadable as hell and nobody will be able to tell at the first glance what makes a password valid. How about this instead?
```
val hasUppercaseChar = password.any { it.isUppercase() }
val hasLowercaseChar = password.any { it.isLowercase() }
val hasDigit = password.any { it.isDigit() }
if(!hasUppercaseChar || !hasLowercaseChar || !hasDigit) {
    // return error
}

// valid
```

Reply:
I agree. Also dislike regex and normally try to avoid it where possible. This is a much more readable approach, will update in a following PR